src/surface_conditions/surface_conditions.jl

```diff
@@ -21,50 +21,151 @@ function update_surface_conditions!(Y, p, t)
     sfc_conditions_values = Fields.field_values(sfc_conditions)
 
+    is_prognostic_sfc = atmos.surface_model isa PrognosticSurfaceTemperature
+    possible_sfc_prognostic_temp =
+        is_prognostic_sfc ? Fields.field_values(Y.sfc.T) : nothing
+
     if sfc_setup isa Function
-        @. sfc_conditions_values = surface_state_to_conditions(
-            sfc_setup(sfc_local_geometry_values.coordinates, int_z_values, t),
-            sfc_local_geometry_values,
-            int_ts_values,
-            projected_vector_data(CT1, int_u_values, int_local_geometry_values),
-            projected_vector_data(CT2, int_u_values, int_local_geometry_values),
-            int_z_values,
-            params,
-            atmos,
-        )
+        if (is_prognostic_sfc)
+            @. sfc_conditions_values = surface_state_to_conditions(
+                sfc_setup(
+                    sfc_local_geometry_values.coordinates,
+                    int_z_values,
+                    t,
+                ),
+                sfc_local_geometry_values,
+                int_ts_values,
+                projected_vector_data(
+                    CT1,
+                    int_u_values,
+                    int_local_geometry_values,
+                ),
+                projected_vector_data(
+                    CT2,
+                    int_u_values,
+                    int_local_geometry_values,
+                ),
+                int_z_values,
+                params,
+                atmos,
+                possible_sfc_prognostic_temp,
+            )
+        else
+            @. sfc_conditions_values = surface_state_to_conditions(
+                sfc_setup(
+                    sfc_local_geometry_values.coordinates,
+                    int_z_values,
+                    t,
+                ),
+                sfc_local_geometry_values,
+                int_ts_values,
+                projected_vector_data(
+                    CT1,
+                    int_u_values,
+                    int_local_geometry_values,
+                ),
+                projected_vector_data(
+                    CT2,
+                    int_u_values,
+                    int_local_geometry_values,
+                ),
+                int_z_values,
+                params,
+                atmos,
+            )
+        end
     elseif sfc_setup isa Fields.Field # This case is needed for the Coupler
         @assert eltype(sfc_setup) <: SurfaceState
         sfc_setup_values = Fields.field_values(sfc_setup)
-        @. sfc_conditions_values = surface_state_to_conditions(
-            sfc_setup_values,
-            sfc_local_geometry_values,
-            int_ts_values,
-            projected_vector_data(CT1, int_u_values, int_local_geometry_values),
-            projected_vector_data(CT2, int_u_values, int_local_geometry_values),
-            int_z_values,
-            params,
-            atmos,
-        )
+        if (is_prognostic_sfc)
```

Comment:
@dennisYatunin Would `sfc_setup` only be a `Field` when running from the coupler? If so I suggest that we ignore `is_prognostic_sfc` in this case, as the coupler should always overwrite surface temperature.